### PR TITLE
Use the range formatter if there's no full range formatter

### LINKED ISSUE #1106: When there's only a documentRangeFormattingProvider, do "full" formatting via that capability
It can be the case that the server advertises only as documentRangeFormattingProvider, but not as documentFormattingProvider.

In that case, still allow the user to format the entire file simply by passing the entire range of the view to the range formatter.

see also https://github.com/microsoft/vscode/issues/99513

### PR DESCRIPTION
**Pull request by rwols**
close #1106

### CODE REVIEW

File: plugin/formatting.py [hunk 1]
```diff
@@ -59,13 +59,23 @@ class LspFormatDocumentCommand(LspTextCommand):
...
+        return super().is_enabled() or bool(self.session(LspFormatDocumentRangeCommand.capability))
+
     def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
         session = self.session(self.capability)
-        file_path = self.view.file_name()
-        if session and file_path:
-            session.send_request(
-                text_document_formatting(self.view),
-                lambda response: apply_response_to_view(response, self.view))
+        if session:
```
Comment (rchl) [inline plugin/formatting.py:67]: Why no `file_name()` check anymore?

File: plugin/formatting.py [hunk 2]
```diff
@@ -59,13 +59,23 @@ class LspFormatDocumentCommand(LspTextCommand):
...
+            # Either use the documentFormattingProvider ...
+            session.send_request(text_document_formatting(self.view), self.on_result)
+        else:
+            session = self.session(LspFormatDocumentRangeCommand.capability)
+            if session:
+                # ... or use the documentRangeFormattingProvider and format the entire range.
+                req = text_document_range_formatting(self.view, sublime.Region(0, self.view.size()))
```
Comment (rchl) [inline plugin/formatting.py:74]: We could change `entire_content_range` to `entire_content_region` and use it here.
One place that uses `entire_content_range` could be changed to convert the region with `region_to_range`.

**Comment by rchl** [comment]
Nice PR number

File: plugin/formatting.py (hunk 1 above)
Comment (rwols) [reply plugin/formatting.py:70]: That's already taken care of in `self.session(...)`.

File: plugin/core/views.py [hunk 3]
```diff
@@ -151,12 +151,16 @@ def text_document_identifier(view: sublime.View) -> Dict[str, Any]:
...
+def entire_region(view: sublime.View) -> sublime.Region:
+    return sublime.Region(0, view.size())
+
+
 def entire_content(view: sublime.View) -> str:
-    return view.substr(sublime.Region(0, view.size()))
+    return view.substr(entire_region(view))
 
 
 def entire_content_range(view: sublime.View) -> Range:
```
Comment (rchl) [inline plugin/core/views.py:162]: Would be more accurate with `_region` suffix.

File: plugin/core/views.py [hunk 4]
```diff
@@ -151,12 +151,16 @@ def text_document_identifier(view: sublime.View) -> Dict[str, Any]:
     return {"uri": uri_from_view(view)}
 
 
+def entire_region(view: sublime.View) -> sublime.Region:
```
Comment (rchl) [inline plugin/core/views.py:154]: Kinda goes out-of-line with naming? `entire_content_region` would match equivalent function better?

Comment (rwols) [reply plugin/core/views.py:154]: Renamed in 62cf41e3b17a91316772fa416a061965a2bd19a9

**Pull request merged**